### PR TITLE
Issue #3: Adds initial setup for aks cluster on Azure

### PR DESCRIPTION
**Pull request by SonOfLope**
This PR is allowing an aks cluster to be created with admin access given to defined ad_groups variable.
It aims to achieve the following :
- [x] : Provide a reusable module for kubernetes allowing deployment in different regions.
- [x] : Provide a reusable virtual network allowing  separation of concerns and reusability.
- [x] : Creates a staging aks cluster for the team.

### CODE REVIEW

File: terraform/staging/main.tf [hunk 1]
```diff
@@ -58,3 +39,77 @@ provider "azurerm" {
...
+  location            = var.location_1
+  resource_group_name = azurerm_resource_group.rg.name
+
+  vnet_name   = "vnet-${var.aks_name}"
+  subnet_name = "subnet-${var.aks_name}"
+
+  address_space           = [var.virtual_network_address]
+  subnet_address_prefixes = [var.subnet_address]
+  tags                    = var.tags
+}
```
Comment (ThomasCardin) [inline terraform/staging/main.tf:54]: Some variables should be public

File: terraform/staging/main.tf [hunk 2]
```diff
@@ -58,3 +39,77 @@ provider "azurerm" {
...
+  network_subnet         = module.cluster-network-0.subnet_name
+
+  service_cidr         = var.service_cidr
+  dns_service_ip       = var.dns_service_ip
+  pod_cidr             = var.pod_cidr
+  storage_account_name = null
+
+  additional_node_pools = var.additional_node_pools
+
+  tags = var.tags
```
Comment (ThomasCardin) [inline terraform/staging/main.tf:101]: Same here

**Review by ThomasCardin — CHANGES_REQUESTED** [review]

**Review by rngadam — APPROVED** [review]
I'm not entirely sure that I could get started deploying my own kubernetes cluster from the README.md. How could we get this to the point where anyone can create a kubernetes cluster on Azure free credits? Also getting queueing someone from Microsoft to review. Can be done in a future issue.

**Comment by rngadam** [comment]
@SonOfLope ...also the description checklist has no checkmarks.

looks very clean, nice naming.

**Comment by SonOfLope** [comment]
> I'm not entirely sure that I could get started deploying my own kubernetes cluster from the README.md. How could we get this to the point where anyone can create a kubernetes cluster on Azure free credits? Also getting queueing someone from Microsoft to review. Can be done in a future issue.

I added some  information in README.md. Currently since we use Devops Azure for the pipeline it would not be an ideal set up for someone to try it so I will open an issue for it and hopefully by then we are no longer using Azure Devops.

**Review by cfiaGeetika — APPROVED** [review]

**Review by ThomasCardin — APPROVED** [review]

**Pull request merged**